### DISCUSSION TITLE
Possible bug - disconnecting from Xdrip more than 20 min, causes permanent unpairing of Dexcom One transmitter

### DISCUSSION THREAD

**Discussion by Nick-Athe** [NightscoutFoundation/xDrip#2655] [Ideas]
I need some help on this!!!! 
It happened to me twice in the past 3 days. My son left away his phone for about 20-30 minutes and from then on it is impossible to re-connect.  When I start pairing again, for the 1st cycle I get the readings but on the 2nd 5-min cycle I get the message "Unpaired".   When disconnected for a short period, say 5 min, it reconnects with no problem. 

![image](https://user-images.githubusercontent.com/124153831/216031139-2ded4272-e3a3-4538-9dba-fa5f015f83a8.png)
![image](https://user-images.githubusercontent.com/124153831/216031247-6a10019f-568e-4c08-b965-270ec705c2f3.png)
![image](https://user-images.githubusercontent.com/124153831/216031289-65cbee48-243f-4495-8d94-2d73a6b90cd6.png)

Ple

**Comment by Nick-Athe** [comment]
FYI: the only solution so far was to use new sensor / transmitter

**Comment by steve8x8** [comment]
This is a KingKong Mini (2? 2Pro?), right? - can you provide some details about the device, the OS running, and the BT settings in xDrip?

**Comment by Nick-Athe** [comment]
Yes, it is king kong mini 2, android 10

**Comment by jamorham** [comment]
Disable the `special pairing workaround` setting (default is off and is for samsung only) and it should be able to re-pair.

**Reply by Nick-Athe** [reply]
I am certain I tried this and restarted phone before initiating a new pairing.   But how is it possible that 1st pairing works alright and 2nd pairing attempt in 5 min leads to unpairing

![image](https://user-images.githubusercontent.com/124153831/216087771-ba3081f8-be31-42b1-bf19-7f6027e1e046.png)![image](https://user-images.githubusercontent.com/124153831/216087881-a3610ee8-d4fa-4605-ae35-e2f465029717.png)

![image](https://user-images.githubusercontent.com/124153831/216087941-f7561c3e-1f67-462e-b1e2-190ae17e7b59.png)

![image](https://user-images.githubusercontent.com/124153831/216088069-df0d29db-6d68-461e-8504-9c6c66143206.png)

**Reply by Navid200** [reply]
Your question is not clear.  You have posted several screenshots showing the incorrect setting you have just been asked to correct.

Would you please disable "Special Pairing Workaround" and see what happens?

If doing that does not restart readings, you can try this to get it going: https://navid200.github.io/xDrip/docs/MissedPairRequest.html

**Reply by Nick-Athe** [reply]
The screenshot you see is not current, I changed it back and forth several times (special pairing workaround). 
I switched back to libre now cos I can't leave my son without a cgm.

I am using also the instruction for MissedPairRequest. 1st cycle / initial pairing all works fine, the second 5 min cycle leads to unpairing

**Reply by Nick-Athe** [reply]
also small disconnections (5-10) worked alright, the problem appears when my son forgot to take with him his phone for about 20-30 min

**Reply by Navid200** [reply]
This is a connectivity issue.  Even if the transmitter is just sitting on your desk, you should still be able to test and verify connectivity.
Please leave the special pairing request disabled.  

Please don't post a screenshot unless there is a very good reason.  A screenshot that is not current does not help.

Would you please post a screenshot of the following page?
Settings &#8722;> Less common settings &#8722;> Bluetooth Settings

**Reply by Nick-Athe** [reply]
I have now connected to libre/miao but left dexcom and transmitter on. Is it possible to revert to same dexcom  again.? How

**Reply by Nick-Athe** [reply]
Please leave the special pairing request disabled. / I already did so Navid, we were talking on fb chat couple of days ago, after changing / trying soke things you suggested to post the problem here

**Reply by Navid200** [reply]
Sorry, I was wrong.  Your son's phone is set up to use Libre and cannot also read from G6.  

It can using a variant.  So, it is not going to be easy.  It requires installing another version of xDrip.  I explain below and you can try.   But, it is understandable if you don't want to.

This explains where you can get variants from:  https://navid200.github.io/xDrip/docs/Variants.html

After downloading a variant, you can install it.   Now, you can run the variant and the original xDri on the same phone at the same time.
You can set up the variant to connect to G6.  And the original xDrip will continue to read from MiuaoMiao.

**Reply by Navid200** [reply]
> Please leave the special pairing request disabled. / I already did so Navid, we were talking on fb chat couple of days ago, after changing / trying soke things you suggested to post the problem here

I am really puzzled from this post.  I am sorry I don't understand you.
You attached a screenshot showing that you have special pairing workaround enabled.
You were asked to disable it.  You posted the screenshot again.   I asked you to disable it and please don't post an outdates screenshot.

Now, you have posted this comment.  Why?

**Reply by Nick-Athe** [reply]
I don't think variants will work for.me as it's a C coded dex1 and I believe the version that works for C is after Januqry 2023

**Reply by Navid200** [reply]
That is incorrect.  Please look at the link I have provided.  You can access all recent Nightly variants there.

**Reply by Nick-Athe** [reply]
Hi Navid, you were right about variants, I installed the xDrip-plus-variant1-20230128-b58ca21.apk and made the settings as in https://navid200.github.io/xDrip/docs/Dexcom_page.html and unchecked the special workaround. 
On the first pairing, all went ok, I got the readings of Dexcom (it was the sensor/transmitter in initialised on Tuesday) but on the second 5 min cycle on the Phone Service State: Unbond Transmitter a. False, Brain State Unbonding Errors 2. On the 1st / initial cycle, on the Classic Status Page the connection status was Dexcon9M authenticated, on the second it is not connected, 
Any ideas what to do?

**Reply by Navid200** [reply]
Yes, please post a screenshot of this page:
Settings −> Less common settings −> Bluetooth Settings

**Reply by Nick-Athe** [reply]
![image](https://user-images.githubusercontent.com/124153831/216350420-8b6f2b05-6012-46d8-afeb-e64f4181c632.png)
![image](https://user-images.githubusercontent.com/124153831/216350448-51d25188-3ea4-4568-8654-066056842c3e.png)
![image](https://user-images.githubusercontent.com/124153831/216350536-7c51a435-a9df-46f4-b6d2-9ccd0e3fd26f.png)

**Reply by Navid200** [reply]
So, those are from the variant that is using G6, right?

Can you also show the following page?  Again, from the variant that uses G6.

Settings &#8722;> Less common settings

**Reply by Nick-Athe** [reply]
Yes, they come from Variant Xdrip for Dex1
![image](https://user-images.githubusercontent.com/124153831/216352329-6c24423f-8c2b-4bb4-bd11-5ae40e8931d7.png)
![image](https://user-images.githubusercontent.com/124153831/216352390-83445b42-ae2e-4c79-a63a-6a6ff548b2c0.png)
![image](https://user-images.githubusercontent.com/124153831/216352436-b05dc063-ea92-482e-b88d-5099e68b5712.png)

**Reply by Navid200** [reply]
Thanks for reporting this.
Let's wait for other suggestions.

**Reply by jamorham** [reply]
@Nick-Athe Please enable OB1 debugging https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector see if the problem reproduces and submit the logs to me using the upload logs feature along with some identifier so I know they come from you and I can take a look. Make sure there isn't another device also trying to connect to the same transmitter.

@Navid200 why is a variant being used here do you know?

**Reply by Navid200** [reply]
@jamorham Nick is using Libre to keep his son's sugar under control.
At the same time, I suggested using a variant so that we can continue to troubleshoot Dexcom One because that is his preference.  
So, we can work on troubleshooting without compromising his son's control.

**Reply by Nick-Athe** [reply]
Thank you. I used the scanner to change configuration. I will leave it running for the night and I will send the log files

**Reply by Nick-Athe** [reply]
I just got the 1st connection and Unpair afterwards 
Marked "from Nick-Athe to jamorham"
I will send more tomorrow 
Thank you

**Reply by Nick-Athe** [reply]
From time to time I get the pop up prompt to pair transmitter. Its random though, no pattern. Sometimes it works but I believe normal operation shouldn't be asking to pair again. 
I confirm no other phone in the house has the Dexcom app.

**Reply by Navid200** [reply]
No, under normal conditions, it only asks you once to pair.  

But, if you have enabled allow OB1 unbonding, you may experience this problem.
That's why that setting should be disabled.

**Reply by Nick-Athe** [reply]
I confirm OB1 Unbonding is NOT checked,

**Reply by Nick-Athe** [reply]
I sent a new log file "from Nick-Athe

**Reply by Nick-Athe** [reply]
> @Nick-Athe Please enable OB1 debugging https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector see if the problem reproduces and submit the logs to me using the upload logs feature along with some identifier so I know they come from you and I can take a look. Make sure there isn't another device also trying to connect to the same transmitter.
> 
> @Navid200 why is a variant being used here do you know?

Hi Jamorham, just checking if you received the log files

**Reply by jamorham** [reply]
I received the logs and I believe that these show that the phone bluetooth framework is losing the pairing or otherwise having problems there.

If you have another android phone I would suggest testing xDrip on that phone with the same transmitter to see if the problem is localized to the phone which I think there is a good chance that is the problem. (Make sure to disable the kingkong collector when testing on another phone or you will get other problems)

For the kingkong you can try to install the latest manufacturer update to see if that helps. It is a mediatek bluetooth chip and normally the support on those is pretty good

**Reply by Nick-Athe** [reply]
I used a Redmi10C, android 11.
Made a new installation of xdrip and deleted the other installation from kingkong
I Checked that all settings are as per document 

1st connection went smoothly and received data. 2nd attempt to connect was unsuccessful, same messages as with KingKong...Unpaired.
Didn't manage to connect again without intervention

**Reply by Nick-Athe** [reply]
Please forgive my technical ignorance, I am just trying to figure out what is the problem. 
The first time I used a Dex1 transmitter, it worked perfectly for 7 days, no disconnections (even if I had, by mistake, the Samsung workarounds checked), great interoperability with AAPS and great improvement of my son's glucose control compared to Libre1. But one day my son left the phone away for 20-30min and from then on I couldn't make the connection stable (message ==> unbinding). On the next day, I scrapped the sensor and transmitter and started with a new sensor and transmitter.  Again for 1.5 days no problem all worked well. But the next morning it got disconnected and from then on again I couldn't have a stable connection.  
1. One it fails, it continues to fail. Is it possible that the transmitter or phone keeps a record of this 1st connection failure? Even if we change the device, the problem persists. I cleared cache and past pairings on phones, and nothing changed.  
2. Is it possible that the problem is with the patch for "C" transmitters and not xDrip? How can we engage the team that decoded "c" transmitters give an opinion on the issue?

**Reply by Navid200** [reply]
I suggested to you to post here because the individual who knows more about this than me is here.  He has already spent time corresponding with you.
We both understand your problem and we both want to help you.

Please be patient.

**Reply by jamorham** [reply]
I think if this were a more widespread problem we would have heard about it by now from elsewhere too. Someone has offered to send me one of these kingkong phones so I will see if I can reproduce the issue based on what you've described. If you still have the packaging, can you provide me with the date which appears next to the factory icon on the original transmitter label?

**Reply by Nick-Athe** [reply]
I think if this were a more widespread problem we would have heard about it by now from elsewhere too. // I see your point. I am watching the local group in Greece and the use of dex1 is still limited, some use the dex app, few others mention disconnections, some managed to connect even before the patch for C was made available. 
Also, some users used to buy them from Bulgaria. Delivery in Greece started mid-December 2022. 
I was thinking that possibly the problem is with kingkong, but trying it on a new Redmi10C - with superior Bluetooth - and having exactly the same issue replicated (failure to reconnect after the 1st successful authentication, lost pairing) made me think that possibly it's about "C" transmitters. 

the date next to the factory icon is 2022-10-05.
I am sending you also the other marks, if of any use: REF STT-DO-001   /  C2259M

Thank you for your support

**Reply by jamorham** [reply]
Overall this does remind me of the problem seen with the Samsung A22. In the end the best solution there was to upgrade the phone to the latest version (Android 12) which resolved issues of random loss of pairing. Is there an update available for the Redmi 10 C and does the kingkong have the latest manufacturer updates as well (Jan 12th update)? edit: my mistake the last mini2 update was Oct 2022.

**Reply by Nick-Athe** [reply]
I ll work on this on my side, update the Redmi first and kingkong afterwards. I didn't do the update yesterday when I saw I was having the same issue with redmi.

**Reply by jamorham** [reply]
It might not be phone related but it is definitely worth performing an update as we saw that resolve problems on the Samsung A22 but the Redmi 10C has qualcomm chipset so it would be strange for them to show the same fault.

**Reply by Nick-Athe** [reply]
I updated to the latest miui 13 on redmi10c. I m not sure if this what you meant by updating or flashing the rom. Nothing changed after miui update

**Reply by Navid200** [reply]
No flashing of the ROM.  Just updating to the latest Android update provided by the manufacturer of your phone.

**Reply by jamorham** [reply]
@Nick-Athe yes, update is what I meant.  On the redmi10c are you having a problem with it unpairing or just failing to get data?

**Reply by Nick-Athe** [reply]
I understand that transmitter receives data, such as the calibrations. Is it possible that it received some data that prevents transmitter to operate properly?  The first time I used dex1 I calibrated the sensor and worked fine for 7days. The second time readings were accurate and didn't callibrate and worked fine for 1 day.

**Reply by Nick-Athe** [reply]
It is the same issue. 
I change the transmitter ID and revert to normal ID to initiate a new pairing. 
Wait for 1-5 min, it asks for pairing. I get the data. 
On the 2nd cycle it fails to get data, it unpairs. Not possible to read again. 

Both Redmi and KingKong run the latest android of manufacturer. 11 and 10 respectively.

**Reply by jamorham** [reply]
Can you explain what you mean about changing the transmitter id? Please describe the steps you are following to help me understand and try to reproduce the error.

**Reply by Nick-Athe** [reply]
sure, see all steps. 

a.	Initial pairing ok, started the sensor without problems. Phone/Dex1 worked ok for few days. 
b.	One day, after leaving the phone away for more than 15-20min, on the Phone Service State State: Unpaired. And Brain State: Unbonding Errors. Pairing is lost, I assume, and cannot be reinstated by using “Restarting collector”. Or any other way. 
During the days it worked ok, in case it lost signal for a few minutes, it would reconnect without any intervention. 
c.	From this incident onward, the only way to get data from the transmitter is by using https://navid200.github.io/xDrip/docs/MissedPairRequest.html (this is what I mean by changing transmitter ID). But with this procedure, I only get data once. I have to repeat the process every time I want to read Dex1.

**Reply by Nick-Athe** [reply]
Is it possible that having dex1 away/disconnected for a prolonged period is perceived as an effort to restart the sensor to extend its life and it gets blocked permanently?

**Reply by Nick-Athe** [reply]
Today I started a new sensor. I used a new transmitter as my previous one doesn't seem to connect. I was wondering if I can send you by post this receiver so you can try to read from it and find out whats going on with C type dex1. I got a feeling that some security feature prevents transmitters from pairing again once disconnected for a reason. Please let me know. I know that there are couple of more people having the same issue with me

**Reply by jamorham** [reply]
please contact me jamorham at gmail if you still want to send the transmitter for testing

**Comment by Navid200** [comment]
Have you disabled battery optimization?

**Reply by Nick-Athe** [reply]
yes, of course

**Reply by Nick-Athe** [reply]
aaps, xdrip, have full access, incl. location, without battery optim.

**Comment by Nick-Athe** [comment]
Good afternoon everyone, 
I plan to start a new dex1 sensor in 3 days (when the current Libre1 expires). 
I am not sure if you managed to run any tests, or get an idea on what causes the issue of unpairing. 
In addition, something I don't really know, is if the transmitter gets to keep in its memory (if any) a record of past failures in connecting, or the device it was connected to. And if this could be a source of failure for subsequent connection attempts to the same or other devices. 
This time I will be using Redmi 10C /Android 11/Miui 13 and not KingKong2 as it "died" suddenly tonight and can't turn it on. 
If you got any suggestions, please let me know. 
Thank you in advance, Nick

**Reply by jamorham** [reply]
Please let us know how it goes.
